src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -1,30 +1,154 @@
+    else:
+        scope = '/subscriptions/' + definitions_client.config.subscription_id
+        if resource_group_name:
+            scope = scope + '/resourceGroups/' + resource_group_name
+
+    role_id = None
+    try:
+        uuid.UUID(role)
+        role_id = role
+    except ValueError:
+        pass
+
+    if not role_id: #retrieve role id
         role_defs = list(definitions_client.list(scope, "roleName eq '{}'".format(role)))
         if not role_defs:
-            raise CLIError('Role {} doesn\'t exist.'.format(role))
+            raise CLIError("Role '{}' doesn't exist.".format(role))
         elif len(role_defs) > 1:
-            raise CLIError('More than one roles match the given name {}'.format(role))
+            ids = [r.id for r in role_defs]
+            err = "More than one roles match the given name '{}'. Please pick a value from '{}'"
+            raise CLIError(err.format(role, ids))
         role_id = role_defs[0].id
 
+    object_id = _get_object_id(assignee)
     properties = RoleAssignmentProperties(role_id, object_id)
     assignment_name = uuid.uuid4()
     return assignments_client.create(scope, assignment_name, properties)
 
+def list_apps(app_id=None, display_name=None, identifier_uri=None, query_filter=None):
+    client = _graph_client_factory()
+
+    sub_filters = []
+    if query_filter:
+        sub_filters.append(query_filter)
+    if app_id:
+        sub_filters.append("appId eq '{}'".format(app_id))
+    if display_name:
+        sub_filters.append("startswith(displayName,'{}')".format(display_name))
+    if identifier_uri:
+        sub_filters.append("identifierUris/any(s:s eq '{}')".format(identifier_uri))
+
+    return client.applications.list(filter=(' and '.join(sub_filters)))
+
+def list_sps(spn=None, display_name=None, query_filter=None):
+    client = _graph_client_factory()
+
+    sub_filters = []
+    if query_filter:
+        sub_filters.append(query_filter)
+    if spn:
+        sub_filters.append("servicePrincipalNames/any(c:c eq '{}')".format(spn))
+    if display_name:
+        sub_filters.append("startswith(displayName,'{}')".format(display_name))
+
+    return client.service_principals.list(filter=(' and '.join(sub_filters)))
+
+def list_users(upn=None, display_name=None, query_filter=None):
+    client = _graph_client_factory()
+
+    sub_filters = []
+    if query_filter:
+        sub_filters.append(query_filter)
+    if upn:
+        sub_filters.append("userPrincipalName eq '{}'".format(upn))
+    if display_name:
+        sub_filters.append("startswith(displayName,'{}')".format(display_name))
+
+    return client.users.list(filter=(' and ').join(sub_filters))
+
+def create_user(user_principal_name, display_name, password, mail_nick_name=None, #pylint: disable=too-many-arguments
+                immutable_id=None, force_change_password_next_login=False):
+    '''
+    :param mail_nick_name: mail alias. default to user principal name
+    '''
+    client = _graph_client_factory()
```

Comment:
>client = _graph_client_factory() [](start = 4, length = 32)

If you provide the same client factory in the generated.py file, it will be passed in to the command (and thus, there is no need to create it inside your handler)